Tools/cases_generator/generators_common.py

```diff
@@ -17,3 +33,143 @@ def write_header(generator: str, source: str, outfile: TextIO) -> None:
 // Do not edit!
 """
     )
+
+
+def emit_to(out: CWriter, tkn_iter: Iterator[Token], end: str) -> None:
```

Comment:
This could be a method on `CWriter`.

Reply:
I thought about that, but it tracks the parentheses in the `tkn_iter` input, which seems out of scope for a writer.